File: forward-system-frontend/src/core/type/ParticipantType.ts
import {ParticipantTypeEnum} from "@/core/enum/ParticipantTypeEnum.ts";

export type ParticipantType = keyof typeof ParticipantTypeEnum;

export function getParticipantTypeRusName(type: ParticipantType): String {
    if (type == ParticipantTypeEnum.HOST) {
        return "Менеджер Хост";
    }
    if (type == ParticipantTypeEnum.CATCHER) {
        return "Менеджер Кетчер";
    }
    if (type == ParticipantTypeEnum.AUTHOR) {
        return "Автор на рассмотрении";
    }
    if (type == ParticipantTypeEnum.MAIN_AUTHOR) {
        return "Взявший в работу автор";
    }
    if (type == ParticipantTypeEnum.DECLINE_AUTHOR) {
        return "Отказавщийся автор";
    }
}

File: src/main/java/by/forward/forward_system/core/enums/ParticipantType.java
package by.forward.forward_system.core.enums;

import by.forward.forward_system.core.jpa.model.OrderParticipantsTypeEntity;
import lombok.Getter;

@Getter
public enum ParticipantType {
    OWNER("OWNER", "Владелец"),
    CATCHER("CATCHER", "Менеджер Кетчер"),
    HOST("HOST", "Менеджер Хост"),
    AUTHOR("AUTHOR", "Автор на рассмотрении"),
    DECLINE_AUTHOR("DECLINE_AUTHOR", "Отказавшийся автор"),
    MAIN_AUTHOR("MAIN_AUTHOR", "Взявший в работу автор"),
    EXPERT("EXPERT", "Эксперт");

    private final String name;

    private final String rusName;

    ParticipantType(String name, String rusName) {
        this.name = name;
        this.rusName = rusName;
    }

    public static ParticipantType byName(String name) {
        for (ParticipantType type : ParticipantType.values()) {
            if (type.name.equals(name)) {
                return type;
            }
        }
        return null;
    }

    public OrderParticipantsTypeEntity toEntity() {
        return new OrderParticipantsTypeEntity(name);
    }
}
